src/system.cpp

```diff
@@ -0,0 +1,57 @@
+#include "system.hpp"
+
+#include "location_service.h"
+#include "Particle.h"
+#include "cli/conio.hpp"
+#include "consts.hpp"
+
+char SYS_deviceID[32];
+
+SystemDesc_t systemDesc, *pSystemDesc = &systemDesc;
+SystemFlags_t systemFlags;
+
+
+static int SYS_initGPS();
+
+
+int SYS_initSys(void)
+{
+    memset(pSystemDesc, 0, sizeof(SystemDesc_t));
+    systemDesc.deviceID = SYS_deviceID;
+    systemDesc.flags = &systemFlags;
+
+    memset(SYS_deviceID, 0, 32);
+    strncpy(SYS_deviceID, System.deviceID(), 31);
+
+    SYS_initGPS();
+    return 1;
+}
+
+/**
+ * @brief Initialization function for GPS 
+ * Ublox gps, handled by @file gps/location_service.cpp
+ * @return int sucsess
```

Comment:
Normally, success might be 1, fail would be zero. Other way to do it would be success is zero, fail is nonzero (easy to check with if statement).  I would not use fail as -1 if 1 is success.